### DISCUSSION TITLE
GitLab Private NPM Package Registry configuration

### DISCUSSION THREAD

**Discussion by stefandoorn** [renovatebot/renovate#16478] [Request Help] [answered]
Hi,

We host some internal NPM packages in the GitLab NPM registry. I've configured `npmrc`, as well with a token to read from the registry. That does seem to work fine. Packages can be installed.

One of our internal packages is now tagged at v1.3.6. In a repository it has a constraint of `^1.3.0`. But, somehow, it doesn't trigger an update. All other public packages do get updates, the repository dashboard is there, etc. 

What I can find from the relevant logs for this repository:

> "deps":[{"depType":"devDependencies","depName":"@group/package","currentValue":"^1.3.0","datasource":"npm","prettyDepType":"devDependency","lockedVersion":"1.3.5","depIndex":0,"warnings":[],"versioning":"npm","currentVersion":"1.3.5","fixedVersion":"1.3.5","updates":[]}

It doesn't recognize the 1.3.6 version being available there, somehow. 

In the log I do see it can access / download certain things:

> https://gitlab.com/api/v4/packages/npm/@group%2Fpackage (GET,200)

In another repository, I've updated it manually to 1.3.6, and there it shows similar as above but not 1.3.5, but 1.3.6. 

From my config:

```
    "npmrc": "@group=https://gitlab.com/api/v4/packages/npm/\n//gitlab.com/api/v4/packages/npm/:_authToken=",
    "hostRules": [
        {
            "hostType": "npm",
            "baseUrl": "https://gitlab.com/api/v4/packages/npm/",
            "token": "{{ secrets.NPM_READ_TOKEN }}"
        }
    ],
```

Then in the `.gitlab-ci.yml` I've added the secret:

```
variables:
  RENOVATE_SECRETS: "{\"NPM_READ_TOKEN\": \"${NPM_READ_TOKEN}\"}"
```

Validated, the `NPM_READ_TOKEN` is present in this repo. The secret also shows in the renovate-runner logfiles as being loaded (with ***).

Going through all the docs, I feel like this way it should be setup. Using this `.npmrc` locally, when running a `npm outdated`, I do see the newer version.

Am I overlooking something here?

**Comment by viceice** [comment]
remove the authtoken part from npmrc

**Reply by stefandoorn** [reply]
Old:

> "npmrc": "@group=https://gitlab.com/api/v4/packages/npm/\n//gitlab.com/api/v4/packages/npm/:_authToken=",

New:

> "npmrc": "@group=https://gitlab.com/api/v4/packages/npm/\n//gitlab.com/api/v4/packages/npm/",

Like this?

**Reply by viceice** [reply]
yes

**Reply by stefandoorn** [reply]
Adjusted that, but it doesn't seem to make a difference. Don't see a difference in the log file either that points me to something.

**Comment by viceice** [comment, marked as answer]
it's probably because renovate doesn't update ranges If the new version matches the range.

please show the package files with updates message, but only the relevant deps

**Comment by stefandoorn** [comment]
Tagged 1.3.7 to be sure everything looked fine from that perspective, then:

`package.json` from the repo:

```
{
  "devDependencies": {
    "@group/package": "^1.3",
  },
}
```

The package:

```
{
  "name": "@group/package",
  "version": "1.3.7",
}
```

Both the `main` branch and the `development` branch in `@group/package` show 1.3.7 in the `package.json` as well. Although I'm not sure that's relevant.

I do see this when it's handling the repository `@group/package`:

```
"packageJsonName":"@group/package","packageFileVersion":"1.3.7","packageJsonType":"library","npmLock":"package-lock.json","managerData":{"yarnZeroInstall":false,"hasPackageManager":false}
```

So, when going through that repository, I do feel it understand it is at 1.3.7.

Then in the repository in which I expect the update:

```
{"depType":"devDependencies","depName":"@group/package","currentValue":"^1.3","datasource":"npm","prettyDepType":"devDependency","lockedVersion":"1.3.5","depIndex":3,"warnings":[],"versioning":"npm","currentVersion":"1.3.5","fixedVersion":"1.3.5","updates":[]}
```

And in the `urls` section:

> https://gitlab.com/api/v4/packages/npm/@group%2Fpackage (GET,200)

When I open that URL, I do find that version in the JSON output:

```
"1.3.7": {
"version": "1.3.7",
}
```

And:

```
"dist-tags": {
"latest": "1.3.7"
}
```

**Reply by viceice** [reply]
https://docs.renovatebot.com/configuration-options/#rangestrategy

as i told before. renovate won't suggest updates for lockfile only by default.

**Reply by stefandoorn** [reply]
Ah, clear. Looking at that one now.. it's set to `auto` right now.

Slightly in doubt what might be the best option here, as I do want these patch updates to be updated. But I would also like to receive updates for minor/majors.

Or, does `:maintainLockFilesWeekly` actually does the upgrade to 1.3.7 once per week and I'm just being impatient for this update to arrive?

**Comment by stefandoorn** [comment]
Thanks, @viceice - I was impatient in combination with that setting :-)
